### DISCUSSION TITLE
How should I run code when Action done

### DISCUSSION THREAD

**Discussion by dvorakchen** [leptos-rs/leptos#2275] [Q&A] [answered]
I have a event function:
```rust
let handle_send_captcha = move |ev: SubmitEvent| {
        ev.prevent_default();
        action.dispatch("params".to_string());
        //  to something when action done
}
```

I trigger it by click a button, and I want to do something after action done. 
by example: an Action do call a http request, if I write some code below action.dispatch, whose code will executed while action is pending. but actually I want execute whose code when Action be done

any good idea?

**Comment by dvorakchen** [comment]
I think I have found a way, directly write your code in closure parameter of create_action, 
```rust
let captcha_valid_action = create_action(|input: &(String, String)| {
        let (email, captcha) = input.clone();
        async move {
            let res = valid_captcha(&email, &captcha).await;
            //  do something here
            logging::log!("end");
            res
        }
    });
```

but I do not know if there is a better idea

**Comment by gbj** [comment, marked as answer]
There are two possibilities:
1) Write it in the async block, as you said
2) Use `create_effect` and track the `.value()` signal of the action
